xunit-autolabeler-v2/ast_parser/core/yaml_utils.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Dict, List, Tuple
+
+from ast_parser.core import constants, polyglot_drift_data
+from ast_parser.lib import file_utils
+
+import yaml
+
+
+def _handle_overwrites(
+    source_methods_json: List[polyglot_drift_data.PolyglotDriftData],
+    root_dir: str
+) -> None:
+    """Handle overwrites in .drift-data.yml files
+
+    The "overwrite" clause is used to denote when a .drift-data.yml
+    entry should overwrite any automatically-detected test methods.
+    If an overwrite entry is present for a given region tag, this
+    method clears all previously-detected tests for snippet methods
+    containing that region tag.
```

Comment:
Might be good to specify that these 3 private methods modifies the passed source_methods_json in each of the function's docstrings.

Reply:
Done.